### ISSUE TITLE
[CLOSED] Creating elements using Javascript does not work.

### ISSUE THREAD

**Issue by ngokevin** [aframevr/aframe#426]
<a href="https://github.com/caseyyee"><img src="https://avatars.githubusercontent.com/u/1560884?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [caseyyee](https://github.com/caseyyee)**
_Monday Aug 31, 2015 at 20:04 GMT_
_Originally opened as https://github.com/aframevr/aframe-core/issues/51_

----

I was not able to dynamically create vr-elements and add them to a scene:

````html
var scene = document.querySelector('vr-scene');
var cube = document.createElement('vr-cube');
cube.setAttribute('x', 0);
cube.setAttribute('y', 0);
cube.setAttribute('z', -100);
scene.appendChild(cube);
````

**Comment by ngokevin** [comment]
<a href="https://github.com/caseyyee"><img src="https://avatars.githubusercontent.com/u/1560884?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [caseyyee](https://github.com/caseyyee)**
_Monday Aug 31, 2015 at 20:48 GMT_

----

The elements are created, but nothing is rendered:

![image](https://cloud.githubusercontent.com/assets/1560884/9590006/f5c0be68-4fe6-11e5-93f1-bb0c5483b634.png)

**Comment by ngokevin** [comment]
<a href="https://github.com/nickdesaulniers"><img src="https://avatars.githubusercontent.com/u/1823839?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [nickdesaulniers](https://github.com/nickdesaulniers)**
_Tuesday Sep 01, 2015 at 17:33 GMT_

----

The existing elements seem to have a loaded event attached to them from vr-markup.js which calls this.render(), but the dynamically created ones do not.

**Comment by ngokevin** [comment]
<a href="https://github.com/dmarcos"><img src="https://avatars.githubusercontent.com/u/39342?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [dmarcos](https://github.com/dmarcos)**
_Thursday Sep 03, 2015 at 15:47 GMT_

----

The problem is that an element attaches itself to the parent after being loaded/initialized. In the case of creating elements in JS the initialization and attachment to the DOM tree are decoupled. The following patch makes an object to attempt adding itself to the parent right after being attached to the DOM:

https://github.com/MozVR/vr-markup/commit/9654a0be3d189a6460dccf549a5767954871283a

This should now as you would expect

**Comment by ngokevin** [comment]
<a href="https://github.com/caseyyee"><img src="https://avatars.githubusercontent.com/u/1560884?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [caseyyee](https://github.com/caseyyee)**
_Saturday Sep 05, 2015 at 01:30 GMT_

----

This is still an issue.   Here is an example of a scene dynamic created using createElement that demonstrates the issue:
http://caseyyee.github.io/vr-markup/examples/dynamic.html

After some investigation with @dmarcos, it looks like we never end up fulfilling the `pendingElements` condition:  https://github.com/MozVR/vr-markup/blob/dev/src/core/vr-scene.js#L108

**Comment by ngokevin** [comment]
<a href="https://github.com/dmarcos"><img src="https://avatars.githubusercontent.com/u/39342?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [dmarcos](https://github.com/dmarcos)**
_Sunday Sep 06, 2015 at 05:53 GMT_

----

This has been fixed on ac063de. When creating elements dynamically the elements load and then added to the scene. vr-scene was waiting for the load event of those elements when it had been already been fired. 

@caseyyee This is [your example working] (http://mozvr.github.io/vr-markup/examples/dynamic.html)
